onnxruntime/core/providers/dml/DmlExecutionProvider/src/BucketizedBufferAllocator.cpp

```diff
@@ -212,15 +212,6 @@ namespace Dml
             ORT_THROW_HR(E_INVALIDARG);
         }
         const auto* allocInfo = static_cast<const AllocationInfo*>(opaqueHandle);
-
-        auto owner = allocInfo->GetOwner();
-        //The owner can be null if the resource was wrapped via CreateGPUAllocationFromD3DResource
-        if (owner != nullptr && owner != this)
-        {
-            // This allocation doesn't belong to this allocator!
-            ORT_THROW_HR(E_INVALIDARG);
```

Comment:
Hmm, is this validation no longer applicable?

Reply:
DML only has one type of allocator at the moment (`BucketizedBufferAllocator`), but this validation means that our internal allocator cannot decode data that was allocated from the external allocator, which there's nothing wrong with. We always return `AllocationInfo` pointers, for both internal and external allocations.